securedrop_client/gui/widgets.py

```diff
@@ -1328,12 +1326,16 @@ def __init__(
         self,
         source: Source,
         sdc_home: str,
-        controller: Client,
+        controller: Optional[Client],
         is_authenticated: bool,
         parent=None
     ) -> None:
         super().__init__(parent)
         self.source = source
+
+        if controller is None:
```

Comment:
If this is optional, but we raise on `None`, shouldn't this just be not optional?

Reply:
ah so this is intentional: making `SourceConversationWrapper.controller` optional would mean some more significant refactoring, i.e. `Window` has `self.controller` as optional (resolving this would touch most of our tests). my intention with this PR was not make a sprawling change and instead just prevent more typing issues from landing on `master`, let's do this in a followup

Reply:
[updated #25 to this effect]

Reply:
huh actually scratch this, looks like it's resolved on master (just rebased on top)